src/lib.rs

```diff
@@ -1,744 +1,345 @@
-    copy_compute_info: libloading::Symbol<
-        'lib,
-        unsafe extern "C" fn(
-            *mut IRShaderReflectionOpaque,
-            version: IRReflectionVersion,
-            csinfo: *mut IRVersionedCSInfo,
-        ) -> bool,
-    >,
-    copy_vertex_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_fragment_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_geometry_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_hull_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    copy_domain_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_compute_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_vertex_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_fragment_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_geometry_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_hull_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    release_domain_info:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    get_resource_count:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-    get_resource_locations:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRShaderReflectionOpaque) -> ()>,
-}
-
-pub struct IRShaderReflection<'lib> {
-    me: *mut IRShaderReflectionOpaque,
-    funcs: IRReflectionFn<'lib>,
-}
-
-impl<'lib> Drop for IRShaderReflection<'lib> {
+use std::{
+    ffi::{CStr, OsStr},
+    mem::MaybeUninit,
+    sync::Arc,
+};
+
+#[allow(
+    non_upper_case_globals,
+    dead_code,
+    non_camel_case_types,
+    non_snake_case,
+    clippy::too_many_arguments,
+    clippy::enum_variant_names
+)]
+mod bindings;
+
+use bindings::IRError;
+pub use bindings::{
+    IRCSInfo_1_0, IRComparisonFunction, IRDescriptorRangeType, IRFilter, IRHitGroupType,
+    IRObjectType, IRRaytracingPipelineFlags, IRReflectionVersion, IRResourceLocation,
+    IRResourceType, IRRootConstants, IRRootParameter1,
+    IRRootParameter1__bindgen_ty_1 as IRRootParameter1_u, IRRootParameterType,
+    IRRootSignatureDescriptor1, IRRootSignatureFlags, IRRootSignatureVersion, IRShaderStage,
+    IRShaderVisibility, IRStaticBorderColor, IRStaticSamplerDescriptor, IRTextureAddressMode,
+    IRVersionedCSInfo, IRVersionedRootSignatureDescriptor,
+    IRVersionedRootSignatureDescriptor__bindgen_ty_1 as IRVersionedRootSignatureDescriptor_u,
+};
```

Comment:
It's a tad unfortunate that half the bindings are wrapped, and the other half is plainly reexported.

Maybe we can expose these inside a (renamed) `bindings` or `ffi` module to keep them a bit more separated to the caller?